### PR TITLE
Fix for #2893 - do not attempt WorkloadIdentityCredential for MSI FIC (for Id.Web v2)

### PR DESCRIPTION
**Pull request by bgavrilMS**
See https://github.com/AzureAD/microsoft-identity-web/pull/2813

### CODE REVIEW

File: src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs [hunk 1]
```diff
@@ -23,19 +23,7 @@ public class ManagedIdentityClientAssertion : ClientAssertionProviderBase
...
-                    WorkloadIdentityClientId = managedIdentityClientId,
-                    ExcludeAzureCliCredential = true,
-                    ExcludeAzureDeveloperCliCredential = true,
-                    ExcludeAzurePowerShellCredential = true,
-                    ExcludeInteractiveBrowserCredential = true,
-                    ExcludeSharedTokenCacheCredential = true,
-                    ExcludeVisualStudioCodeCredential = true,
-                    ExcludeVisualStudioCredential = true
-                });
+            _credential = new ManagedIdentityCredential(managedIdentityClientId);
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs:26]: What's the point of using `ManagedIdentityCredential` and not MSAL.NET directly?

**Review by jmprieur — COMMENTED** [review]
I still don't understand.

File: src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs:26]: It reduces churn. I will do it on id.web 3.x

**Comment by bgavrilMS** [comment]
@jmprieur @jennyf19 - gentle ping on this.

File: src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs (hunk 1 above)
Comment (jmprieur) [inline src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs:26]: did we decide to use ManagedIdentityCredentials, and not directly MSAL.NET ?

Comment (bgavrilMS) [reply src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs:26]: I think either way is fine. Let me push an update to use MSAL.

Comment (bgavrilMS) [reply src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs]: Done, pls have a look.

**Comment by jennyf19** [comment]
why not cherry-pick the other commit?

**Comment by bgavrilMS** [comment]
> why not cherry-pick the other commit?

It is subtly different, because in 3.x we introduced a breaking change to the base classes. Have a look at https://github.com/AzureAD/microsoft-identity-web/commit/e3a059f3576afe0a87dae7de66aff59a50c11c41#diff-d515a839836ec1adc2d63d93fb6c38698476023fa4b444ca4f62424f8a3fa19aR56

**Review by jmprieur — APPROVED** [review]
LGTM
We should run a release build ...

**Pull request merged**